types/express-serve-static-core/index.d.ts

```diff
@@ -61,7 +61,7 @@ export interface RequestHandler<
         req: Request<P, ResBody, ReqBody, ReqQuery, LocalsObj>,
         res: Response<ResBody, LocalsObj>,
         next: NextFunction,
-    ): void;
+    ): void | Promise<void>;
```

Comment:
It's unfortunate that this is an interface; if this were a union, this would be valid:

```ts
type RequestHandler<
    P = ParamsDictionary,
    ResBody = any,
    ReqBody = any,
    ReqQuery = ParsedQs,
    LocalsObj extends Record<string, any> = Record<string, any>,
> =
  | ((
        req: Request<P, ResBody, ReqBody, ReqQuery, LocalsObj>,
        res: Response<ResBody, LocalsObj>,
        next: NextFunction,
    ) => void)
  | ((
        req: Request<P, ResBody, ReqBody, ReqQuery, LocalsObj>,
        res: Response<ResBody, LocalsObj>,
        next: NextFunction,
    ) => Promise<void>)
  ;
```

It being an interface seems to have been a side effect of something back in TS <2.2, which is mega far in the past.

Then again, the new form has sort of caught everyone who was more or less mistakenly returning something from a handler thinking that meant anything, so, this PR is probably okay as-is.